tests/transport/test_send_recv.cpp

```diff
@@ -0,0 +1,355 @@
+#include <iostream>
+#include <iomanip>
+#include <ghex/threads/none/primitives.hpp>
+#include <ghex/transport_layer/message_buffer.hpp>
+#include <ghex/transport_layer/shared_message_buffer.hpp>
+#include <ghex/common/timer.hpp>
+#include <gtest/gtest.h>
+
+#ifdef GHEX_TEST_USE_UCX
+#include <ghex/transport_layer/ucx/context.hpp>
+using transport = gridtools::ghex::tl::ucx_tag;
+#else
+#include <ghex/transport_layer/mpi/context.hpp>
+using transport = gridtools::ghex::tl::mpi_tag;
+#endif
+
+using threading = gridtools::ghex::threads::none::primitives;
+using context_type = gridtools::ghex::tl::context<transport, threading>;
+using communicator_type = typename context_type::communicator_type;
+
+#define NITERS 100000
+
+
+template<typename MsgType, typename CommType>
+auto test_ring_send_recv_ft(CommType& comm, MsgType &&rmsg, MsgType &&smsg) 
```

Comment:
since you are testing futures:
```suggestion
auto test_ring_send_recv_ft(CommType& comm, MsgType& rmsg, MsgType& smsg) 
```